### DISCUSSION TITLE
Listening to events

### DISCUSSION THREAD

**Discussion by KZelab** [katspaugh/wavesurfer.js#3289] [Q&A] [answered]
Hello, I am trying to listen to the events for mousing over and leaving while using the region plugin. This is what I am attempting 
![image](https://github.com/katspaugh/wavesurfer.js/assets/69273221/a04e022c-10da-4595-adcc-f9f9201f8080)

None of these have worked as expected, I am trying to update my app to the latest version of wavesurfer 

Thank you and best wishes

**Comment by katspaugh** [comment, marked as answer]
The particular events you're interested in are only exposed on individual Region objects.

So it should be something like:

```js
const region = RegionsPlugin.addRegion({
  start: 0,
  end: 8,
})

region.on('over', onMouseOver)

region.on('leave', onMouseLeave)
```

Examples: https://wavesurfer.xyz/examples/?regions.js
Docs: https://wavesurfer.xyz/docs/types/plugins_regions.RegionEvents

**Reply by KZelab** [reply]
That makes so much sense
Thank you Katpaugh
